mirgecom/eos.py

```diff
@@ -699,7 +707,7 @@ def gas_const(self, cv: ConservedVars,  # type: ignore[override]
             ($\rho$), energy ($\rho{E}$), momentum ($\rho\vec{V}$), and the vector
             of species masses, ($\rho{Y}_\alpha$).
         """
-        y = cv.species_mass_fractions
+        y = species_mass_fractions if cv is None else cv.species_mass_fractions
```

Comment:
Kind of yuck.   What about just making species_mass_fractions a required argument and passing cv.species_mass_fractions instead of CV when possible?

Reply:
I added a warn to deprecate this... Since it involve BC and initializer, let's give an extra time before changing this.